pkg/controllers/remediation/eventhandlers_test.go

```diff
@@ -0,0 +1,372 @@
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+				},
+			},
+			wantQLen: 0,
+		},
+		{
+			name: "update event: equal cluster condition",
+			args: args{
+				operation: "Update",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+				oldObj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+			},
+			wantQLen: 0,
+		},
+		{
+			name: "update event: not equal cluster condition",
+			args: args{
+				operation: "Update",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionFalse,
+							},
+						},
+					},
+				},
+				oldObj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+					Status: clusterv1alpha1.ClusterStatus{
+						Conditions: []metav1.Condition{
+							{
+								Type:   "Ready",
+								Status: metav1.ConditionTrue,
+							},
+						},
+					},
+				},
+			},
+			wantQLen: 1,
+		},
+		{
+			name: "generic event",
+			args: args{
+				operation: "Generic",
+				q:         &controllertest.Queue{Interface: workqueue.New()},
+				obj: &clusterv1alpha1.Cluster{
+					ObjectMeta: metav1.ObjectMeta{Name: "member1"},
+				},
+			},
+			wantQLen: 1,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			queue := tt.args.q
+			h := newClusterEventHandler()
+			switch tt.args.operation {
+			case "Create":
+				createEvent := event.CreateEvent{Object: tt.args.obj}
+				h.Create(context.TODO(), createEvent, queue)
+			case "Delete":
+				deleteEvent := event.DeleteEvent{Object: tt.args.obj}
+				h.Delete(context.TODO(), deleteEvent, queue)
+			case "Update":
+				updateEvent := event.UpdateEvent{ObjectNew: tt.args.obj, ObjectOld: tt.args.oldObj}
+				h.Update(context.TODO(), updateEvent, queue)
+			case "Generic":
+				genericEvent := event.GenericEvent{Object: tt.args.obj}
+				h.Generic(context.TODO(), genericEvent, queue)
+			default:
+				t.Errorf("no support operation %v", tt.args.operation)
+				return
+			}
+
+			if got := queue.Len(); got != tt.wantQLen {
+				t.Errorf("clusterEventHandler %v() process queue length = %v, want %v", tt.args.operation, got, tt.wantQLen)
```

Comment:
One more thing, actually we don't need to log the target function name here, as the go test would print it out, like:
```
=== RUN   Test_remedyEventHandler
=== RUN   Test_remedyEventHandler/create_event:_remedy_with_clusterAffinity
=== RUN   Test_remedyEventHandler/create_event:_remedy_with_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_old_and_new_remedy_all_have_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_one_of_the_old_and_new_remedy_have_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/update_event:_the_old_and_new_remedy_clusterAffinity_changed
=== RUN   Test_remedyEventHandler/delete_event:_the_remedy_with_clusterAffinity
=== RUN   Test_remedyEventHandler/delete_event:_the_remedy_with_nil_clusterAffinity
=== RUN   Test_remedyEventHandler/generic_event
--- PASS: Test_remedyEventHandler (0.00s)
    --- PASS: Test_remedyEventHandler/create_event:_remedy_with_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/create_event:_remedy_with_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_old_and_new_remedy_all_have_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_one_of_the_old_and_new_remedy_have_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/update_event:_the_old_and_new_remedy_clusterAffinity_changed (0.00s)
    --- PASS: Test_remedyEventHandler/delete_event:_the_remedy_with_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/delete_event:_the_remedy_with_nil_clusterAffinity (0.00s)
    --- PASS: Test_remedyEventHandler/generic_event (0.00s)
```
It's no big deal, we didn't pay attention to this before.  clean it or not, it's up to you.

Reply:
That makes sense. I'll change it.